examples/typescript/yourCoin/index.ts

```diff
@@ -0,0 +1,142 @@
+/* eslint-disable no-console */
+/* eslint-disable max-len */
+
+import { Account, AccountAddress, Aptos, AptosConfig, Network, NetworkToNetworkName } from "@aptos-labs/ts-sdk";
+import { compilePackage, getPackageBytesToPublish } from "../utils";
+
+/**
+ * This example demonstrate how one can publish a new custom coin to chain.
+ * It uses the MoonCoin.move module that can be found in this folder
+ *
+ * Before running this example, we should compile the package locally:
+ * 1. Acquire the Aptos CLI, see https://aptos.dev/cli-tools/aptos-cli/use-cli/install-aptos-cli
+ * 2. cd `~/aptos-ts-sdk/examples/typescript`
+ * 3. Run `pnpm run your_coin`
+ */
+
+const MOON_COINS_TO_MINT = 100;
+const MOON_COINS_TO_TRANSFER = 100;
+
+// Setup the client
+const APTOS_NETWORK: Network = NetworkToNetworkName[process.env.APTOS_NETWORK] || Network.DEVNET;
+const config = new AptosConfig({ network: APTOS_NETWORK });
+const aptos = new Aptos(config);
+
+/** Register the receiver account to receive transfers for the new coin. */
+async function registerCoin(receiver: Account, coinTypeAddress: AccountAddress): Promise<string> {
+  const transaction = await aptos.build.transaction({
+    sender: receiver.accountAddress,
+    data: {
+      function: "0x1::managed_coin::register",
+      typeArguments: [`${coinTypeAddress.toString()}::moon_coin::MoonCoin`],
+      functionArguments: [],
+    },
+  });
+
+  const senderAuthenticator = aptos.sign.transaction({ signer: receiver, transaction });
+  const pendingTxn = await aptos.submit.transaction({ transaction, senderAuthenticator });
+
+  return pendingTxn.hash;
+}
+
+/** Transfer the newly created coin to a specified receiver address */
+async function transferCoin(
+  sender: Account,
+  receiverAddress: AccountAddress,
+  amount: number | bigint,
+): Promise<string> {
+  const transaction = await aptos.build.transaction({
+    sender: sender.accountAddress,
+    data: {
+      function: "0x1::aptos_account::transfer_coins",
+      typeArguments: [`${sender.accountAddress.toString()}::moon_coin::MoonCoin`],
```

Comment:
You can drop the `toString()` and it will automatically do `toString()` for objects that are implemented.